### ISSUE TITLE
Agenda toggle knob with a function

### ISSUE THREAD

**Issue by elliscwc** [wix/react-native-calendars#937]
This might related to https://github.com/wix/react-native-calendars/issues/309

I want to show the calendar list with a call of the function in the agenda component, is there a way to do it correctly? Someone has suggested this way but is no longer working
```
function Agenda(props) {
  const agenda = useRef(null);

  useEffect(() => {
    // doesn't work anymore
    agenda.current.setScrollPadPosition(0, true);
    agenda.current.enableCalendarScrolling();
    
    // suggestion?
    const animated = false;
    agenda.current.toggleKnob(animated);
  }, []);

  return (
    <Agenda
      ref={agenda}
      items={[]}
      renderItem={() => <View />}
      renderEmptyDate={() => <View />}
      renderEmptyData={() => <View />}
      rowHasChanged={(r1, r2) => r1.text !== r2.text}
      refreshing={false}
      style={styles.container}
    />
  );
}
```
![3](https://user-images.githubusercontent.com/29953232/65513705-562f7e00-df0e-11e9-8085-c9f24141951c.png)

**Comment by jraleman** [comment]
This will be a cool feature to have!

**Comment by Inbal-Tish** [comment]
@elliscwc Do you mean open/close Agenda's calendar programmatically?

**Comment by elliscwc** [comment]
Yes, @Inbal-Tish. it would be helpful when it can be activated via other buttons on the screen!

**Comment by flobady** [comment]
Hello, I was able to solve my problem with this:
` <Button 
title="knob"
onPress={() => {
          let initPos = refAgenda.current.initialScrollPadPosition()
          refAgenda.current.setScrollPadPosition(initPos, true)}
        }>`

**Comment by waynedunkley** [comment]
For those that find this post and are looking for a solution when using the `ExpanableCalendar` within a `CalendarProvider` as such:
```
<CalendarProvider date={current} onDateChanged={onDateChanged}>
  <ExpandableCalendar
    ref={ref}
    {...props}
  />
</CalendarProvider>
```
My solution was to use the expandableCalendar ref and mock a dayPress
`calendarRef.current.contentRef.onDayPress({ dateString: '2021-03-20' })`

**Comment by Brayden-kwak** [comment]
> const agenda = useRef(null);

could you show me full code that I can fully understand? 
I don't understand where the 'agendaRef' came from. 
Thanks

**Comment by Brayden-kwak** [comment]
> Actually to open and close programmatically the `Agenda` thanks to the knob i use the method `toggleCalendarPosition()`:
> 
> ```
> <TouchableOpacity
>   onPress={() => {
>     agendaRef.current.toggleCalendarPosition(true); // or false if you want to close
>   }}>
>   <Text>Open/close</Text>
> </TouchableOpacity>
> ```

could you show me full code that I can fully understand? 
I don't understand where the 'agendaRef' came from. 
Thanks

**Comment by lbenz** [comment]
@Brayden-kwak, agendaRef is a reference to the Agenda component that is rendered, take a look at this sample:

`
import React, {useRef} from "react";
import {Text, TouchableOpacity} from "react-native";

function Agenda(props) {
    const agendaRef = useRef(null);
    
    return (
        <>
            <TouchableOpacity
                onPress={() => {
                    agendaRef.current.toggleCalendarPosition(true); // or false if you want to close
                }}>
                <Text>Open/close</Text>
            </TouchableOpacity>

            <Agenda
                ref={agendaRef}
                items={[]}
            />
        </>
    )
}
`

**Comment by Brayden-kwak** [comment]
Cool! Thanks! 
Can I apply it to the AgendaList? 
I'm stuck to apply .svg icon to the AgnedaList with expandable slender.
